crates/uv-install-wheel/src/wheel.rs

```diff
@@ -801,6 +826,37 @@ pub(crate) fn parse_scripts(
     scripts_from_ini(extras, python_minor, ini)
 }
 
+/// Rename a file with a fallback to copy that switches over on the first failure.
+#[derive(Default, Copy, Clone)]
+enum RenameOrCopy {
+    #[default]
+    Rename,
+    Copy,
+}
+
+impl RenameOrCopy {
+    /// Try to rename, and on failure, copy.
+    ///
+    /// Usually, source and target are on the same device, so we can rename, but if that fails, we
+    /// have to copy. If renaming failed once, we switch to copy permanently.
+    fn rename_or_copy(&mut self, from: impl AsRef<Path>, to: impl AsRef<Path>) -> io::Result<()> {
+        match self {
+            Self::Rename => match fs_err::rename(from.as_ref(), to.as_ref()) {
+                Ok(()) => {}
+                Err(err) => {
```

Comment:
Same question here: can we make this branch exclusive to "different drive"? Is there an error kind for it?